test/components/generators/chat/test_azure.py

```diff
@@ -60,6 +60,27 @@ def test_init_with_parameters(self, tools):
         assert component.tools == tools
         assert component.tools_strict
         assert component.azure_ad_token_provider is not None
+        assert component.max_retries == 5
+
+    def test_init_with_0_max_retries(self, tools):
```

Comment:
I would add a short docstring explaining the purpose of this test.
Something like: tests that the `max_retries` init param is taken into account if not null